lib/driver.js

```diff
@@ -5,26 +5,78 @@
 
 const {LDKeyPair} = require('crypto-ld');
 const X25519KeyPair = require('x25519-key-pair');
+const {constants: securityConstants} = require('security-context');
 
 /**
- * Returns a `did:key` method DID Document for a given DID.
+ * Returns a `did:key` method DID Document for a given DID, or a key document
+ * for a given DID URL (key id).
  * Note: This is async only to match the async `get()` signature of other
  * `did-io` drivers.
  *
- * @param {string} did
- * @returns {Promise<DidDocument>}
+ * Either a `did` or `url` param is required:
+ * @param {string} [did] - DID URL or a key id (either an ed25519 key or an
+ *   x25519 key-agreement key id).
+ * @param {string} [url] - alias for the `did` url param, supported for better
+ *   readability of invoking code.
+ *
+ * Usage:
+ *
+ *   ```
+ *   await driver.get({did}); // -> did document
+ *   await driver.get({url: keyId}); // -> public key node
+ *   ```
+ *
  * @throws Unsupported Fingerprint Type.
  * @throws Cannot get - missing DID.
+ * @returns {Promise<DidDocument|object>} Resolves to a DID Document or a
+ *   public key node with context.
  */
-async function get({did} = {}) {
+async function get({did, url} = {}) {
+  did = did || url;
   if(!did) {
     throw new Error('Cannot get - missing DID.');
```

Comment:
This would better follow our standard practice:

```suggestion
    throw new TypeError('"did" must be a string.');
```

Reply:
@dlongley thanks, applied!